swap/src/protocol/bob/state.rs

```diff
@@ -31,19 +31,19 @@ pub enum BobState {
         #[serde_as(as = "DisplayFromStr")]
         change_address: bitcoin::Address,
     },
-    SwapSetupCompleted(State2),
-    BtcLocked(State3),
+    SwapSetupCompleted(Box<State2>),
+    BtcLocked(Box<State3>),
     XmrLockProofReceived {
-        state: State3,
+        state: Box<State3>,
         lock_transfer_proof: TransferProof,
         monero_wallet_restore_blockheight: BlockHeight,
     },
-    XmrLocked(State4),
-    EncSigSent(State4),
-    BtcRedeemed(State5),
-    CancelTimelockExpired(State6),
-    BtcCancelled(State6),
-    BtcRefunded(State6),
+    XmrLocked(Box<State4>),
+    EncSigSent(Box<State4>),
+    BtcRedeemed(Box<State5>),
+    CancelTimelockExpired(Box<State6>),
+    BtcCancelled(Box<State6>),
+    BtcRefunded(Box<State6>),
```

Comment:
Did clippy tell you to do this? :)

I wouldn't bother, this part is not super performance critical and we certainly should not optimize this without having measured an improvement.